Assets/Mirror/Tests/Editor/NetworkManagerTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿using NUnit.Framework;
+using UnityEngine;
+
+namespace Mirror.Tests
+{
+    [TestFixture]
+    public class NetworkManagerTest
+    {
+        GameObject gameObject;
+        NetworkManager manager;
+
+        [SetUp]
+        public void SetupNetworkManager()
+        {
+            gameObject = new GameObject();
+            manager = gameObject.AddComponent<NetworkManager>();
+        }
+
+        [TearDown]
+        public void TearDownNetworkManager()
+        {
+            GameObject.DestroyImmediate(gameObject);
+        }
+
+        [Test]
+        public void VariableTest()
+        {
+            Assert.That(manager.dontDestroyOnLoad == true);
+            Assert.That(manager.runInBackground == true);
+            Assert.That(manager.startOnHeadless == true);
+            Assert.That(manager.showDebugMessages == false);
+            Assert.That(manager.serverTickRate == 30);
+            Assert.That(manager.offlineScene == "");
+            Assert.That(manager.networkAddress == "localhost");
+            Assert.That(manager.maxConnections == 4);
+            Assert.That(manager.autoCreatePlayer == true);
+            Assert.That(manager.spawnPrefabs.Count == 0);
+            Assert.That(manager.numPlayers == 0);
+            Assert.That(manager.isNetworkActive == false);
+
+            Assert.That(NetworkManager.networkSceneName == "");
+            Assert.That(NetworkManager.startPositionIndex == 0);
+            Assert.That(NetworkManager.startPositions.Count == 0);
+        }
+
+        [Test]
+        public void OnValidateTest()
+        {
+            Assert.That(gameObject.GetComponent("NetworkServer") == null);
+        }
+
+        [Test]
+        public void StartServerTest()
+        {
+            manager.StartServer();
+
+            Assert.That(manager.isNetworkActive == true);
+            Assert.That(manager.mode == NetworkManagerMode.ServerOnly);
```

Comment:
this is nice.
maybe add one more check to make sure that NetworkServer.active is true after calling networkmanager.startserver?